jmclient/jmclient/wallet_utils.py

```diff
@@ -353,31 +353,49 @@ def wallet_display(wallet, gaplimit, showprivkey, displayall=False,
     else:
         return walletview
 
-def cli_password_check():
-    password = get_password('Enter wallet encryption passphrase: ')
-    password2 = get_password('Reenter wallet encryption passphrase: ')
+def cli_get_wallet_passphrase_check():
+    password = get_password('Enter wallet file encryption passphrase: ')
+    password2 = get_password('Reenter wallet file encryption passphrase: ')
     if password != password2:
         print('ERROR. Passwords did not match')
-        return False, False
-    password_key = btc.bin_dbl_sha256(password)
-    return password, password_key
+        return False
+    return password
 
-def cli_get_walletname():
+def cli_get_wallet_file_name():
     return raw_input('Input wallet file name (default: wallet.json): ')
 
-def cli_user_words(words):
-    print('Write down this wallet recovery seed\n\n' + words +'\n')
-
-def cli_user_words_entry():
-    return raw_input("Input 12 word recovery seed: ")
-
-def persist_walletfile(walletspath, default_wallet_name, encrypted_seed,
-                       callbacks=(cli_get_walletname,)):
+def cli_display_user_words(words, mnemonic_extension):
+    text = 'Write down this wallet recovery mnemonic\n\n' + words +'\n'
+    if mnemonic_extension:
+        text += '\nAnd this mnemonic extension: ' + mnemonic_extension + '\n'
+    print(text)
+
+def cli_user_mnemonic_entry():
+    mnemonic_phrase = raw_input("Input 12 word mnemonic recovery phrase: ")
+    mnemonic_extension = raw_input("Input mnemonic extension, leave blank if there isnt one: ")
```

Comment:
I think it's a good idea to post-process `mnemonic_extension` itself with `strip()` ; as an experiment i entered the ME with a trailing space in the recovery workflow and it of course recovers the wrong wallet.